src/macro_expansion.jl

```diff
@@ -78,12 +78,19 @@ struct MacroExpansionError
     context::Union{Nothing,MacroContext}
     ex::SyntaxTree
     msg::String
+    "The source position relative to the node - may be `:begin` or `:end` or `:all`"
     position::Symbol
+    "Error that occurred inside the macro function call (note that this may not be defined)"
```

Comment:
Can we just set this to `nothing` rather than leaving it `#undef`?

Reply:
Yeah. The reason why I went with `#undef` was that theoretically, `nothing` could also be `throw`n.. But honestly, that probably never be useful in practice, and `#undef` can be a bit tricky itself, so I'm fine with bringing in the sentinel value.

Reply:
I agree it's a theoretical possibility that we loose a tiny amount of information by doing this... but yeah `#undef` really is a bit of a pain to use correctly :sweat_smile: